File: 240322/최고의 33위치/best-place-of-33.py
n = int(input())

grid = [
    list(map(int, input().split()))
    for _ in range(n)
]

result = 0

def count_area(x, y):
    count = 0
    for i in range(x, x+3):
        for j in range(y, y+3):
            if grid[i][j] == 1:
                count += 1

    return count



for i in range(n-2):
    for j in range(n-2):
        result = max(result, count_area(i, j))

print(result)

File: 241003/최고의 33위치/best-place-of-33.cpp
#include <iostream>
using namespace std;

int n;
int grid[20][20];

int getGold(int row, int col){
    int count = 0;

    for(int i = row; i <= row+2 ; i++){
        for(int j = col; j <= col+2; j++){
            count += grid[i][j];
        }
    }
    return count ;

}

int main() {
    // 여기에 코드를 작성해주세요.


    int result = 0;

    cin >> n;
    for(int i = 0; i < n; i++ )
        for(int j = 0; j<n; j++)
            cin >> grid[i][j];

    
    int count = 0;

    for(int i = 0; i < n-2 ; i++){
        for(int j = 0; j < n-2 ; j++){

            
            result = getGold(i, j);
            // cout << i << " " << j << " " << result << endl; 
            count = max(result, count );

        }
    }

    cout << count;
    





    return 0;
}